Additional fixes to Zed lake design doc (#2595)
brimdata/zed: docs/lake/design.md

> # Zed Lake Design

Author: It seemed like saying "Doc" in the title was redundant. :man\_shrugging:

## Data Pools

> A lake is composed of _data pools_.  Each data pool is organized

Author: "Comprised of" appeared many times through the doc, and was flagged by the grammar checker, so I've changed it to the convention I learned about via [https://www.grammarly.com/blog/comprise-vs-compose/.](https://www.grammarly.com/blog/comprise-vs-compose/.) :man\_shrugging:  :smile:

Reviewer: Or get fancy: `A like comprises _data pools_.`

Each journal entry implies a snapshot of the data in a pool.  A snapshot
is computed by applying the ADD/DROP directives in sequence from entry TAIL to
the journal entry in question, up to HEAD.  This gives the set of commit tags

> that comprise a snapshot.  A snapshot may then be assembled by scanning,

Author: "Scanned by scanning" seemed redundant, so I picked a different leading word. I also considered "retrieved". :man\_shrugging:

> a file for exclusive access and checking that it has zero length after
> a successful open.

> Second, strong read/write ordering semantics (as exists in Amazon S3)

Author: "read/writer" seemed like an odd pairing.

## Lake Introspection

> Commit history, metadata about segments, segment key spaces,

Author: We used the single word "metadata" everywhere else in the doc, and that seems to be the standard convention. :man\_shrugging:

> Commit history, metadata about segments, segment key spaces,

Author: We used the single word "metadata" everywhere else in the doc, and that seems to be the standard convention. :man\_shrugging:

## Keyless Data

This is TBD.  Data without a key should be accepted some way or another.

> One approach could be to simply assign the "zero-value" as the pool key,
> or a configured default value could be used.  This would make key-based

Reviewer: Opening with "One approach" really begs for parallel construction in the next clause.
```suggestion
One approach is to simply assign the "zero-value" as the pool key; another
is to use a configured default value.  This would make key-based
```
